src/pool.rs

```diff
@@ -1071,4 +1077,103 @@ mod test {
             "Unexpected error: {claim_err}"
         );
     }
+
+    #[tokio::test]
+    async fn dropping_pool_closes_all_connections() {
```

Comment:
This test is almost the same as from your PR, though I removed `DecrementOnDrop` as you suggested